### ISSUE TITLE
Get Actions for Miele devices

### ISSUE THREAD

**Issue by thetrebor** [astrandb/miele#232]
### Feature request

So if i'm reading the docs properly you can pass in a device ID to the /actions and /programs GET apis and get a list of the device programs which you would need for making custom cards.  It's frustratingly hard to get this data outside of HASS but i can't seem to nudge the python scripts in the plugin to let me just do an authenticated GET

curl -X 'GET' \
  'https://api.mcs3.miele.com/v1/devices/000176461151/programs' \
  -H 'accept: application/json'

Would it be possible to add to the Services dropdown the ability to just do this GET on an ID so we can see the programs that are available on each device?

### What version of the integration are you using?

miele-0-1-9

**Comment by astrandb** [comment]
If you look in the diagnostic download for your devices you can see if anything useful is reported on the /programs endpoint. Please test with the device in different states, such as power on, power off, delayed start enabled etc.
Personally I don't have access to any devices with program features so I cannot do any testing myself.
When I did my initial research I noted that the implementation in the API was not very consequent, to say the least.
If you can find out what and when the API reports reliably, we can have a discussion how to use that information in the integration.
There is no more documentation from Miele on this topic than what is published on their developer pages. I have asked...

**Comment by thetrebor** [comment]
Yeah it appears the programs end point just returns [] or a 500 error.
The diagnostics similarly. 

```
"programs": {
      "httpStatus": 500
    },
```

```
"programs": [],
```
```
{
        "appliance": "Microwave combination oven",
        "key": "stateProgramId",
        "raw": 0,
        "localized": ""
      },
      {
        "appliance": "Microwave combination oven",
        "key": "stateProgramType",
        "raw": 0,
        "localized": "Program"
      },
      {
        "appliance": "Microwave combination oven",
        "key": "stateProgramPhase",
        "raw": 0,
        "localized": ""
      },
```
and the logs are not helpful in that they don't display the ID used, just that one was used.

so is the only way to reverse engineer this to feed it program ids one at a time and see what it does?

**Comment by thetrebor** [comment]
just to follow up I have a ticket in with Miele about the chaotic API responses.  they're responding, but slowly

**Comment by AlexCherrypi** [comment]
I also would really appreciate this.
I would love to be able to select the program, temperature, TwinDos setting, max spinning speed, starting time etc. for my Miele washing machine. 
I would love to completely ditch the miele app in favor of having everything accessible via homeassistant.
